src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -30,53 +37,108 @@
+     * @see <a href="https://unix.stackexchange.com/questions/139191/whats-the-difference-between-primary-selection-and-clipboard-buffer/139193#139193">Longer text over clipboards</a>
      */
-    public void setContent(ClipboardContent content) {
-        clipboard.setContent(content);
+    public static void addX11Support(TextInputControl input) {
+        input.selectedTextProperty().addListener((observable, oldValue, newValue) -> {
+            if (!newValue.isEmpty()) {
+                primary.setContents(new StringSelection(newValue), null);
+            }
+        });
+        input.setOnMouseClicked(event -> {
+            if (event.getButton() == MouseButton.MIDDLE) {
+                input.insertText(input.getCaretPosition(), getContentsPrimary());
+            }
+        });
     }
 
     /**
-     * Get the String residing on the clipboard.
+     * Get the String residing on the system clipboard.
      *
-     * @return any text found on the Clipboard; if none found, return an
-     * empty String.
+     * @return any text found on the Clipboard; if none found, return an empty String.
      */
-    public String getContents() {
+    public static String getContents() {
         String result = clipboard.getString();
         if (result == null) {
             return "";
         }
         return result;
     }
 
+    /**
+     * Get the String residing on the primary clipboard.
+     *
+     * @return any text found on the primary Clipboard; if none found, try with the system clipboard.
+     */
+    public static String getContentsPrimary() {
+        Transferable contents = primary.getContents(null);
+        if (contents != null && contents.isDataFlavorSupported(DataFlavor.stringFlavor)) {
+            try {
+                return (String) contents.getTransferData(DataFlavor.stringFlavor);
+            } catch (UnsupportedFlavorException | IOException e) {
+                LOGGER.warn(e.getMessage());
+            }
+        }
+        return getContents();
+    }
+
+    /**
+     * Puts content onto the system clipboard.
+     *
+     * @param content the ClipboardContent to set as current value of the system clipboard.
+     */
+    public void setContent(ClipboardContent content) {
+        clipboard.setContent(content);
+        setPrimaryClipboardContent(content);
+    }
+
+    /**
+     * Puts content onto the primary clipboard.
+     *
+     * @param content the ClipboardContent to set as current value of the primary clipboard.
+     */
+    public void setPrimaryClipboardContent(ClipboardContent content) {
+        primary.setContents(new StringSelection(content.getString()), null);
+    }
+
     public void setHtmlContent(String html) {
         final ClipboardContent content = new ClipboardContent();
         content.putHtml(html);
         clipboard.setContent(content);
+        setPrimaryClipboardContent(content);
```

Comment:
Is this line really necessary? I think, only when selecting text by the mouse should copy something. Not in all cases.

Maybe, I understand the code wrong. Currently, it reads that the content is copied into the system clipboard and the "primary" one (which is the one where the middle mouse pastes from)

Reply:
It is not necessary. I have included that call because in the original issue (#2389) the JabRef contributor said:

> in the "Copy DOI URL" function which copies the URL into the XA_CLIPBOARD selection (**and not (also) into XA_PRIMARY**)

So at the beginning I thought that we wanted to copy to both clipboards a specific content, for example, when a user do a click in the `Copy Version` button. But we could remove it, if you prefer that behavior.

Reply:
OK, then leave it in.